SalesforceCommerceCloud/commerce-sdk-isomorphic
# Add SLAS social login helper

In order to implement social login in the PWA Kit and enable this feature for our customers, we need a social login helper in `commerce-sdk-isomorphic`.

GUS: [W-16543984](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001zRfsQYAS/view)

## Changes
- Add a SLAS login helper for IDP (social) login
  - This was split into 2 helpers: `authorizeIDP` and `loginIDPUser` BECAUSE:
    - The `authorizeCustomer` returns an auth URL that has to be navigated to from the top level browser in order for the app to be properly redirected to the 3rd party login page and for the user to be able to input their login credentials
    - Once the 3rd party login is complete, the retail app will call `loginIDPUser` to exchange the `usid` and `code` for an access token
- Add `apiPaths` readonly map to the client template that maps the API names to the API path

## Testing
- Run `yarn test` and verify that all tests pass

E2e test:
- Go to https://wasatch-mrt-yuna.mrt-storefront-staging.com/ in an **incognito window**
- Click on the login modal
- Open **Chrome DevTools** and navigate to the **Network** tab
- Click on **Login with Google**
- Verify that you are redirected to the Google login page
- Once login is complete, you should be redirected back to the women's tops page (I have just set this as the redirect URL for now)
- Verify that the `token` call is in the Network tab and you see the access token in the response
